src/tools/compiletest/src/runtest.rs

```diff
@@ -1336,7 +1337,24 @@ actual:\n\
                                 .map(|s| s.to_string()));
                 }
             }
+            MirOpt => {
+                args.extend(["-Z",
+                             "dump-mir=all",
+                             "-Z"]
+                            .iter()
+                            .map(|s| s.to_string()));
 
+
+                let mir_dump_dir = self.get_mir_dump_dir();
+                if !mir_dump_dir.exists() {
+                    fs::create_dir(mir_dump_dir.clone()).expect("the dir should exist");
```

Comment:
I think this should be the "racy" create-dir option, probably-- though I guess each test will be making disjoint directories, so maybe it's not needed.